var/spack/repos/builtin/packages/chlorop/package.py

```diff
@@ -0,0 +1,58 @@
+##############################################################################
+# Copyright (c) 2013-2016, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/llnl/spack
+# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+import distutils.dir_util
+import os
+import tempfile
+
+
+class Chlorop(Package):
+    """Chlorop predicts the presence of chloroplast transit peptides
+    in protein sequences and the location of potential cTP cleavage
+    sites. You will need to obtain the tarball by visiting the
+    URL and completing the form. You can then either run spack
+    install with the tarball in the directory, or add it to a
+    mirror."""
+
+    homepage = "http://www.cbs.dtu.dk/services/ChloroP/"
+
+    version('1.1',
+            'eb0ba6b28dfa735163ad5fc70e30139e46e33f6ae27f87666a7167a4ac5f71d9')
+
+    depends_on('gawk', type='run')
+    patch('chlorop.patch')
+
+    # get tarball from current directory
+    def url_for_version(self, version):
+        return "file://{0}/chlorop-{1}.Linux.tar.gz".format(os.getcwd(),
+                                                            version)
+
+    def install(self, spec, prefix):
+        os.rename('chlorop', 'bin/chlorop')
+        distutils.dir_util.copy_tree(".", prefix)
+
+    def setup_environment(self, spack_env, run_env):
+        run_env.set('CHLOROP', self.spec.prefix)
+        run_env.set('CHLOROTMP', tempfile.gettempdir() + '/chlorop')
```

Comment:
Probably not a good idea, no. Is there a good way to instruct the resultant module file to get a username or uid? Or do you think it would be better to just leave that as something to be set by the user?